File: Sales_Recon/src/dplyr/2-1_Join_Customer_PN.R
library(dplyr)
library(readr)
library(stringr)
library(here)


# Path
path <- here("Sales_Recon")


# Functions
main <- function() {
  # Filenames
  input_1 <- here(path, "output", "1-1. Tax invoice all.csv")
  input_2 <- here(path, "output", "1-2. SAP billing summary.csv")
  output_file <- here(path, "output", "2-1. Join Customer PN.csv")

  # Read data
  df_1 <- read_csv(input_1, col_types = cols(.default = col_character()))
  df_2 <- read_csv(input_2, col_types = cols(.default = col_character()))

  # Process data
  selected_columns <- c("고객명", "sold_to_party", "customer_pn_rev")
  df_1 <- df_1 |> select(selected_columns)
  df_2 <- df_2 |> select(selected_columns)

  df <- full_join(df_1, df_2, by = selected_columns) |>
    distinct() |> # Unique values
    tidyr::drop_na() # Remove missing values

  # Write data
  write_excel_csv(df, output_file) # for 한글 표시
  print("A file is created")
}

main()

File: Sales_Recon/src/pandas/2-1_Join_Customer_PN.py
import pandas as pd
from pathlib import Path


# Path
path = Path(__file__).parent.parent.parent


# Functions
def main():
    # Filenames
    input_1 = path / "output" / "1-1. Tax invoice all.csv"
    input_2 = path / "output" / "1-2. SAP billing summary.csv"
    output_file = path / "output" / "2-1. Join Customer PN.csv"

    # Read data
    df_1 = pd.read_csv(input_1, dtype=str)
    df_2 = pd.read_csv(input_2, dtype=str)

    # Process data
    ## select columns
    selected_columns = ["고객명", "sold_to_party", "customer_pn_rev"]
    df_1 = df_1.loc[:, selected_columns]
    df_2 = df_2.loc[:, selected_columns]

    df = (
        pd.merge(df_1, df_2, on=selected_columns, how="outer")
        .drop_duplicates()  # Unique values
        .dropna()  # Remove missing values
    )

    # Write data
    df.to_csv(output_file, index=False, encoding="utf-8-sig")  # for 한글 표시
    print("A file is created")


if __name__ == "__main__":
    main()
